### PR TITLE
Named agg tuples Issue #18220

### LINKED ISSUE #18220: DOC: DataFrame.groupy.agg with a list of tuples
#### Code Sample, a copy-pastable example if possible

```python
>>> df = pd.DataFrame({'key' : ['a', 'a', 'b', 'b', 'a'],
                       'data' : np.random.randn(5)})

>>> df.groupby('key')['data'].agg([('foo', 'mean')])
          foo
key          
a   -0.944872
b    0.893212

```
#### Problem description
I know its deprecated to rename columns with a dictionary of dictionaries in a groupby but a list of tuples also renames columns.

This behavior is not found in the docs. I accidentally discovered this a while back and then saw it again in Python for Data Analysis.

### PR DESCRIPTION
**Pull request by Ememobong28**
- [ ] closes #18220
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/user_guide/groupby.rst [hunk 1]
```diff
@@ -770,6 +770,77 @@ no column selection, so the values are just the functions.
        max_height="max",
    )
 
+Passing a List of Tuples
```
Comment (mroeschke) [inline doc/source/user_guide/groupby.rst:773]: This can be removed

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -1334,6 +1334,29 @@ def unique(self) -> Series:
 
 
 class DataFrameGroupBy(GroupBy[DataFrame]):
+    groupby_agg_with_tuples_example = """
+            Demonstrates using the `agg` method with a list of tuples for grouping and aggregation.
+
+            Consider a DataFrame `df`:
+
+            key | data
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:1342]: Can you just add a code example of this in `_agg_examples_doc` below?

Comment (Ememobong28) [reply pandas/core/groupby/generic.py:1342]: Okay

File: doc/source/user_guide/groupby.rst (hunk 1 above)
Comment (Ememobong28) [reply doc/source/user_guide/groupby.rst:773]: Okay

File: doc/source/user_guide/groupby.rst [hunk 3]
```diff
@@ -770,6 +770,7 @@ no column selection, so the values are just the functions.
        max_height="max",
    )
 
+
```
Comment (mroeschke) [inline doc/source/user_guide/groupby.rst:773]:
```suggestion
```

File: pandas/core/groupby/generic.py [hunk 4]
```diff
@@ -1424,6 +1447,30 @@ class DataFrameGroupBy(GroupBy[DataFrame]):
...
+    key | data
+    ----|------
+    'a' | 0.5
+    'a' | -1.0
+    'b' | 2.0
+    'b' | -0.5
+    'a' | 1.2
+
+    Example: Applying multiple aggregations - mean and standard deviation
+    df_result = df.groupby('key')['data'].agg([('mean_value', 'mean'), ('std_deviation', 'std')])
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:1466]: Could you format this as a code example like the others in this string?

File: pandas/core/groupby/generic.py [hunk 5]
```diff
@@ -1334,6 +1334,29 @@ def unique(self) -> Series:
 
 
 class DataFrameGroupBy(GroupBy[DataFrame]):
+    groupby_agg_with_tuples_example = """
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:1337]: This isn't needed anymore

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Comment by Ememobong28** [comment]
Okay noted

**Pull request closed without merging**